src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -12,56 +12,69 @@
 namespace Symfony\Component\HttpKernel\Controller;
 
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\ControllerMetadata\Argument\ArgumentMetadataFactoryInterface;
 
 /**
- * Responsible for the creation of the action arguments.
+ * Responsible for the resolving of arguments passed to an action.
  *
- * @author Fabien Potencier <fabien@symfony.com>
+ * @author Iltar van der Berg <kjarli@gmail.com>
  */
-class ArgumentResolver implements ArgumentResolverInterface
+final class ArgumentResolver implements ArgumentResolverInterface
 {
     /**
-     * {@inheritdoc}
+     * @var ArgumentMetadataFactoryInterface
      */
-    public function getArguments(Request $request, $controller)
-    {
-        if (is_array($controller)) {
-            $r = new \ReflectionMethod($controller[0], $controller[1]);
-        } elseif (is_object($controller) && !$controller instanceof \Closure) {
-            $r = new \ReflectionObject($controller);
-            $r = $r->getMethod('__invoke');
-        } else {
-            $r = new \ReflectionFunction($controller);
-        }
+    private $argumentMetadataFactory;
+
+    /**
+     * @var ArgumentValueResolverInterface[]
+     */
+    private $argumentValueResolvers;
 
-        return $this->doGetArguments($request, $controller, $r->getParameters());
+    /**
+     * @param ArgumentMetadataFactoryInterface $argumentMetadataFactory
+     * @param ArgumentValueResolverInterface[] $argumentValueResolvers
+     */
+    public function __construct(ArgumentMetadataFactoryInterface $argumentMetadataFactory = null, array $argumentValueResolvers = [])
```

Comment:
should be `array()` instead of `[]` for consistency with the rest of the code base.